nncf/common/quantization/quantizer_propagation/solver.py

```diff
@@ -0,0 +1,1211 @@
+                quant_prop_graph.unify_pq_scales(primary_pq, pq)
+
+        cloned_prop_quantizers = prop_quantizers_to_process if did_clone else None
+
+        pqs_and_paths = zip(paths, prop_quantizers_to_process)
+        for path, prop_quantizer in pqs_and_paths:
+            status = self.check_transition_via_path(prop_quantizer, path, quant_prop_graph,
+                                                    cloned_prop_quantizers)
+            if status == TransitionStatus.SHOULD_NOT_TRANSITION:
+                if did_clone and prop_quantizer is not curr_prop_quantizer:
+                    quant_prop_graph.remove_propagating_quantizer(prop_quantizer,
+                                                                  keep_propagating_quantizer_at_current_node=True)
+                else:
+                    prop_quantizer = quant_prop_graph.backtrack_propagation_until_accepting_location(prop_quantizer)
+                    if prop_quantizer is not None:
+                        self._finished_propagating_quantizers.append(prop_quantizer)
+            elif status == TransitionStatus.SHOULD_TRANSITION:
+                prop_quantizer = quant_prop_graph.propagate_quantizer_via_path(prop_quantizer, path)
+                surviving_prop_quantizers.append(prop_quantizer)
+            elif status == TransitionStatus.SHOULD_MERGE:
+                # The surviving quantizer will have its "affected edges" set extended
+                # by the corresponding set of the merged quantizer. The assumption
+                # here is that the surviving quantizer should never have to cross
+                # such a "merge point" while backtracking to an accepting location.
+
+                quant_prop_graph.merge_quantizer_into_path(prop_quantizer, path)
+            elif status == TransitionStatus.SHOULD_WAIT_FOR_MERGE:
+                branching_node_key = None
+                for from_node_key, _ in path:
+                    if len(list(quant_prop_graph.successors(from_node_key))) > 1:
+                        branching_node_key = path[0][0]
+                        break
+                assert branching_node_key is not None
+                # pylint:disable=line-too-long
+                self._quantizers_waiting_for_branch_merge.add_propagating_quantizer_to_wait_on_node_key(prop_quantizer,
+                                                                                                        branching_node_key)
+                surviving_prop_quantizers.append(prop_quantizer)
+
+        for prop_quantizer in surviving_prop_quantizers:
+            self._active_propagating_quantizers_queue.appendleft(prop_quantizer)
+        return quant_prop_graph
+
+
+    def get_allowed_quantizer_configs_for_operator(self, quant_det_id: OperatorMetatype) -> List[QuantizerConfig]:
+        return self._operator_allowed_qconfigs_map[quant_det_id]
+
+    def set_allowed_quantization_types_for_operator_nodes(self, quant_prop_graph: QuantizerPropagationStateGraph):
```

Comment:
Could you please check that all public methods have the docstring? If not, add them, please.

Reply:
Done